tests/unittest_builder.py

```diff
@@ -132,12 +133,55 @@ def function(
             __name__,
         )
         function = astroid["function"]
-        # XXX discussable, but that's what is expected by pylint right now
+        # XXX discussable, but that's what is expected by pylint right now, similar to ClassDef
         self.assertEqual(function.fromlineno, 3)
         self.assertEqual(function.tolineno, 5)
         self.assertEqual(function.decorators.fromlineno, 2)
         self.assertEqual(function.decorators.tolineno, 2)
 
+    @staticmethod
+    def test_decorated_class_lineno() -> None:
+        code = textwrap.dedent(
+            """
+        class A:
+            ...
+
+        @decorator
+        class B:
+            ...
+
+        @deco1
+        @deco2(
+            var=42
+        )
+        class C:
+            ...
+        """
+        )
+
+        ast_module: nodes.Module = builder.parse(code)  # type: ignore[assignment]
+
+        # XXX discussable, but that's what is expected by pylint right now, similar to FunctionDef
```

Comment:
I'd say we can remove the `discussable` here? Python 3.8+ has clearly made a choice and I think we have made a choice in `pylint` as well.
I think this "discussion" has been solved.

Reply:
👌🏻 Yeah, for `ClassDef` I think it is resolved. Although to be fair, pylint was fine with either option.
I'll leave the comment for `FunctionDef.fromlineno` in place though, since this still differs from the stdlib.